packages/typespec-azure-portal-core/src/decorators.ts

```diff
@@ -52,6 +55,72 @@ export function $browse(context: DecoratorContext, target: Model, options: Model
   }
 }
 
+export function $promotion(context: DecoratorContext, target: Model, options: Model) {
+  const { program } = context;
+  validateDecoratorUniqueOnNode(context, target, $promotion);
```

Comment:
nit: Since we are retiring `@service{version}` and will only rely on `@versioned`, it only rely on enum versions, the validation of each value provided can be greatly simplified. Only need to validate the enum specified is same as the one specified in `@versioned` on the namespace. Compiler will ensure the values are correct.

Reply:
talked to Allen offline, it is okay to keep this logic since customers are still using the service logic!